packages/x-data-grid/src/components/toolbar/GridToolbarColumnsButton.tsx

```diff
@@ -69,9 +69,9 @@ const GridToolbarColumnsButton = React.forwardRef<HTMLButtonElement, GridToolbar
           aria-expanded={isOpen}
           aria-controls={isOpen ? columnPanelId : undefined}
           startIcon={<rootProps.slots.columnSelectorIcon />}
-          {...buttonProps}
           onClick={showColumns}
           {...rootProps.slotProps?.baseButton}
+          {...buttonProps}
```

Comment:
hey @KenanYusuf i am seeing that this causes the showColumns function to not being called if the onClick is defined by the consumer

meaning the popover is not displayed to the user